assets/js/googlesitekit/widgets/util/combine-widgets.test.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Widgets combination utilities tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { combineWidgets } from './combine-widgets';
+import { getWidgetLayout } from './get-widget-layout';
+import { WIDTH_GRID_CLASS_MAP } from './constants';
+import { WIDGET_WIDTHS } from '../datastore/constants';
+import ReportZero from '../../../components/ReportZero';
+import ActivateModuleCTA from '../../../components/ActivateModuleCTA';
+import CompleteModuleActivationCTA from '../../../components/CompleteModuleActivationCTA';
+
+describe( 'combineWidgets', () => {
+	const getQuarterWidget = ( slug ) => ( { slug, width: WIDGET_WIDTHS.QUARTER } );
+	const getHalfWidget = ( slug ) => ( { slug, width: WIDGET_WIDTHS.HALF } );
+
+	const getRegularState = () => null;
+	const getReportZeroState = ( moduleSlug ) => ( { Component: ReportZero, metadata: { moduleSlug } } );
+	const getActivateModuleCTAState = ( moduleSlug ) => ( { Component: ActivateModuleCTA, metadata: { moduleSlug } } );
+	const getCompleteModuleActivationCTAState = ( moduleSlug ) => ( { Component: CompleteModuleActivationCTA, metadata: { moduleSlug } } );
```

Comment:
This is somewhat related to my response in https://github.com/google/site-kit-wp/pull/2538#discussion_r559516608. I'm not sure we should change it because these tests match the same ones in Storybook and thus are using "real" UI components, which I think better clarifies the intent. Maybe we could add another test case at the top that uses "random" components, but that also wouldn't really improve coverage.